### PR TITLE
Add reference for HEAD request.

### PR DESCRIPTION
**Pull request by lognaturel**
Adds information and clarifies naming. No impact on behavior.

#### What has been done to verify that this works as intended?
I did the variable replacement with Android Studio's refactor tool. I verified that the code still compiles.

#### Why is this the best possible solution? Were any other approaches considered?
I think adding the link provides useful context. I wasn't sure what the head request was for when I previously was in this code. Was reading the spec and saw the description. I also had to debug the code to figure out what endpoint the head request was on so that's why I renamed the variable to clarify that it's the submission URL (server/submission).

#### Are there any risks to merging this code? If so, what are they?
Someone could feel that it's over documenting.

#### Do we need any specific form for testing your changes? If so, please attach one.

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/tasks/InstanceServerUploader.java [hunk 1]
```diff
@@ -160,17 +160,19 @@ private boolean uploadOneSubmission(String urlString, String id, String instance
...
-            // we need to issue a head request
-            HttpHead httpHead = WebUtils.createOpenRosaHttpHead(u);
+            // Issue a head request to confirm the server is an OpenRosa server and see if auth
+            // is required
+            // http://docs.opendatakit.org/openrosa-form-submission/#extended-transmission-considerations
+            HttpHead httpHead = WebUtils.createOpenRosaHttpHead(submissionUri);
 
             // prepare response
-            HttpResponse response = null;
+            HttpResponse response;
```
Comment (dcbriccetti) [inline collect_app/src/main/java/org/odk/collect/android/tasks/InstanceServerUploader.java]: Making this a blank final will simplify understanding the code.

File: collect_app/src/main/java/org/odk/collect/android/tasks/InstanceServerUploader.java [hunk 2]
```diff
@@ -131,27 +131,27 @@ private boolean uploadOneSubmission(String urlString, String id, String instance
 
         File instanceFile = new File(instanceFilePath);
         ContentValues cv = new ContentValues();
-        Uri u = Uri.parse(urlString);
+        Uri submissionUri = Uri.parse(urlString);
```
Comment (dcbriccetti) [inline collect_app/src/main/java/org/odk/collect/android/tasks/InstanceServerUploader.java:134]: Making this final will let readers know the reference will not change.

**Review by dcbriccetti — APPROVED** [review]
Looks fine to me. No need to wait for my `final` suggestions to be implemented, if you’re eager to get this in.

File: collect_app/src/main/java/org/odk/collect/android/tasks/InstanceServerUploader.java (hunk 2 above)
Comment (lognaturel) [reply collect_app/src/main/java/org/odk/collect/android/tasks/InstanceServerUploader.java:134]: Ah, good ideas, @dcbriccetti! Am on it now.

Comment (dcbriccetti) [reply collect_app/src/main/java/org/odk/collect/android/tasks/InstanceServerUploader.java:134]: Never underestimate the cognitive load–reducing power of the `final`! 😁 
Especially over long distances.
I usually don’t use them over short distances.

File: collect_app/src/main/java/org/odk/collect/android/tasks/InstanceServerUploader.java (hunk 1 above)
Comment (dcbriccetti) [reply collect_app/src/main/java/org/odk/collect/android/tasks/InstanceServerUploader.java]: As amazing as IDEA is, I’m surprised it doesn’t offer a change to a blank final for these redundant initializations.

File: collect_app/src/main/java/org/odk/collect/android/tasks/InstanceServerUploader.java (hunk 2 above)
Comment (lognaturel) [reply collect_app/src/main/java/org/odk/collect/android/tasks/InstanceServerUploader.java:134]: This one potentially does change if there is a redirect received from the HEAD request.

Comment (dcbriccetti) [reply collect_app/src/main/java/org/odk/collect/android/tasks/InstanceServerUploader.java:134]: Whoops, I missed the error message from A.S. when I tried it. Sorry.

**Comment by lognaturel** [comment]
👍 Made `response` `final` and improved commits.

**Review by dcbriccetti — APPROVED** [review]

**Review by grzesiek2010 — APPROVED** [review]

**Review by yanokwa — APPROVED** [review]

**Pull request merged**